src/ConfigHandler.cpp

```diff
@@ -23,16 +23,49 @@ void ConfigHandler::setConfigFilePath(string filepath) {
   mConfigFilePath = filepath;
 }
 
-TResult<string> ConfigHandler::getValue(string section, string key) {
-  bool a_bIsUtf8 = false;        // use OS native encoding
-  bool a_bUseMultiKey = false;   // don't support duplicated keys
-  bool a_bUseMultiLine = false;  // don't support multiline values for a key
+/* @brief Returns value of a key as a string
+ */
+TResult<string> ConfigHandler::getValueString(string section, string key) {
+  bool const isUtf8 = false;        // use OS native encoding
+  bool const useMultiKey = false;   // don't support duplicated keys
+  bool const useMultiLine = false;  // don't support multiline values for a key
 
-  CSimpleIniA ini(a_bIsUtf8, a_bUseMultiKey, a_bUseMultiLine);
+  CSimpleIniA ini(isUtf8, useMultiKey, useMultiLine);
   SI_Error rc = ini.LoadFile(mConfigFilePath.c_str());
   if (rc < 0)
     return Error(ErrorCode::FileNotFound,
-                 "ConfigHandler.getValue: couldn't load file");
+                 "ConfigHandler.getValueString: couldn't load file");
 
   return ini.GetValue(section.c_str(), key.c_str());
 }
+
+/* @brief Returns value of a key as integer
+ * @details The following restrictions are given with respect to the format
+ * of a key.
+ * [MainParams]
+ * goodFormat=4711
+ * wrongFormat1=xx4711
+ * wrongFormat2=4711xx
+ * wrongFormat3=47xx11
+ */
+TResult<int> ConfigHandler::getValueInt(string section, string key) {
+  bool const isUtf8 = false;        // use OS native encoding
+  bool const useMultiKey = false;   // don't support duplicated keys
+  bool const useMultiLine = false;  // don't support multiline values for a key
+
+  CSimpleIniA ini(isUtf8, useMultiKey, useMultiLine);
+  SI_Error rc = ini.LoadFile(mConfigFilePath.c_str());
```

Comment:
is it needed to load the file with every `getValue`?

Reply:
`setConfigFilePath` could be a nice place to load the file once and store an instance of `CSimpleIniA` as class member